modules/seqexec/engine/src/main/scala/seqexec/engine/UserEvent.scala

```diff
@@ -7,78 +7,61 @@ import cats.effect.IO
 import cats.implicits._
 import fs2.Stream
 import gem.Observation
-import seqexec.model.{ ClientID, Observer, UserDetails }
+import seqexec.model.ClientID
+import seqexec.model.UserDetails
 
 /**
   * Events generated by the user.
   */
-sealed trait UserEvent[+D <: Engine.Types] {
+sealed trait UserEvent[D<:Engine.Types] {
   def user: Option[UserDetails]
   def username: String = user.foldMap(_.username)
 }
 
-final case class Start(id: Observation.Id, user: Option[UserDetails], clientId: ClientID) extends UserEvent[Nothing]
-final case class Pause(id: Observation.Id, user: Option[UserDetails]) extends UserEvent[Nothing]
-final case class CancelPause(id: Observation.Id, user: Option[UserDetails]) extends UserEvent[Nothing]
-final case class Load(id: Observation.Id, sequence: Sequence) extends UserEvent[Nothing] {
-  val user: Option[UserDetails] = None
-}
-final case class Unload(id: Observation.Id) extends UserEvent[Nothing] {
-  val user: Option[UserDetails] = None
-}
-final case class Breakpoint(id: Observation.Id, user: Option[UserDetails], step: Step.Id, v: Boolean) extends UserEvent[Nothing]
-final case class SkipMark(id: Observation.Id, user: Option[UserDetails], step: Step.Id, v: Boolean) extends UserEvent[Nothing]
-//final case class SetOperator(name: Operator, user: Option[UserDetails]) extends UserEvent[Nothing]
-final case class SetObserver(id: Observation.Id, user: Option[UserDetails], name: Observer) extends UserEvent[Nothing]
-//final case class SetConditions(conditions: Conditions, user: Option[UserDetails]) extends UserEvent[Nothing]
-//final case class SetImageQuality(iq: ImageQuality, user: Option[UserDetails]) extends UserEvent[Nothing]
-//final case class SetWaterVapor(wv: WaterVapor, user: Option[UserDetails]) extends UserEvent[Nothing]
-//final case class SetSkyBackground(wv: SkyBackground, user: Option[UserDetails]) extends UserEvent[Nothing]
-//final case class SetCloudCover(cc: CloudCover, user: Option[UserDetails]) extends UserEvent[Nothing]
-final case class Poll(clientId: ClientID) extends UserEvent[Nothing] {
+final case class Start[D<:Engine.Types](id: Observation.Id, user: Option[UserDetails], clientId: ClientID, f: D#StateType => Boolean) extends UserEvent[D]
```

Comment:
`f` is a user provided check. It is currently used to test resource conflict. I am not sure it was the best mechanism.
Other options I thought of:
1. Use a `GetState` message to inspect the current state, and produce a `Start` event if there was no resource conflict. This is not feasible, because it risk that another event slips in between the processing of the `GetState` and the `Start`. The state check and the sequence status change to "Running" must be done in the same cycle. Another issue is how to report that the sequence was not started because of resource conflicts. Maybe produce a `Busy` event?
2. Remove the event `Start` and make the method `Engine.start` public, with a signature similar to `Engine.load`, that could be used from the function inside a `ModifyState` message. That keeps the state check and the sequence start in the same cycle.